### PR TITLE
Cluster restore from backup

### PR DESCRIPTION
**Pull request by aswietlik**

### CODE REVIEW

File: artifacts/backup-cronjob.yaml [hunk 1]
```diff
@@ -26,23 +26,19 @@ spec:
...
             - "-c"
             - |
               set -ex
 
               # Clone data from master.
-              mkdir /mysql/backup/{{ .Name }}/${JOB_NAME}
-              ncat --recv-only {{ .Spec.Cluster }}-$((0)).{{ .Spec.Cluster }} 3307 | xbstream -x -C /mysql/backup/{{ .Name }}/${JOB_NAME}
+              DATE=`date '+%Y-%m-%d-%H-%M'`
+              mkdir /mysql/backup/{{ .Name }}/${DATE}
+              ncat --recv-only {{ .Spec.Cluster }}-$((0)).{{ .Spec.Cluster }} 3307 | xbstream -x -C /mysql/backup/{{ .Name }}/${DATE}
```
Comment (earlgreyz) [inline artifacts/backup-cronjob.yaml]: `{{ .Spec.Cluster }}-$((0))` is the same as `{{ .Spec.Cluster }}-0`

File: artifacts/mysql-statefulset.yaml [hunk 2]
```diff
@@ -66,6 +79,11 @@ spec:
           subPath: mysql
         - name: conf
           mountPath: /etc/mysql/conf.d
+        {{ if ne .Spec.FromBackup.BackupName "" }}
```
Comment (earlgreyz) [inline artifacts/mysql-statefulset.yaml]: I'm not sure if all `containers` need the backup volume mounted, `initContainers` do the job of restoring the data.

File: artifacts/backup-cronjob.yaml (hunk 1 above)
Comment (aswietlik) [reply artifacts/backup-cronjob.yaml]: Done

File: artifacts/mysql-statefulset.yaml (hunk 2 above)
Comment (aswietlik) [reply artifacts/mysql-statefulset.yaml]: You're right, we only need that in mysql-init.

**Pull request merged**